File: mobile-old/ios/TimeSlot.swift
import Foundation

/// TimeSlot helper computes the discrete token rotation window for a given Unix time.
///
/// Spec (v2):
///   time_slot = floor(unix_time / 15)
///
/// The function operates on seconds since Unix epoch (TimeInterval).
enum TimeSlotHelper {
    private static let defaultWindowSeconds: TimeInterval = 15.0

    static func compute(unixTime: TimeInterval, windowSeconds: TimeInterval = defaultWindowSeconds) -> Int64 {
        precondition(windowSeconds > 0)
        return Int64(floor(unixTime / windowSeconds))
    }
}

File: mobile-old/android/TimeSlot.kt
package com.hnnp.device

import kotlin.math.floor

/**
 * TimeSlot computes the discrete token rotation window for a given Unix time.
 *
 * Spec (v2):
 *   time_slot = floor(unix_time / 15)
 *
 * This helper operates on seconds (Unix time). Callers using milliseconds must
 * convert to seconds first.
 */
object TimeSlot {

    private const val DEFAULT_WINDOW_SECONDS: Double = 15.0

    fun compute(unixTimeSeconds: Double, windowSeconds: Double = DEFAULT_WINDOW_SECONDS): Long {
        require(windowSeconds > 0.0) { "windowSeconds must be > 0" }
        return floor(unixTimeSeconds / windowSeconds).toLong()
    }
}

